core/src/main/java/hudson/PluginManager.java

```diff
@@ -1604,9 +1609,7 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
                     element("dependencies", dependencies);
             new UpdateSite(UpdateCenter.ID_UPLOAD, null).new Plugin(UpdateCenter.ID_UPLOAD, cfg).deploy(true);
             return new HttpRedirect("../updateCenter");
-        } catch (IOException e) {
-            throw e;
-        } catch (Exception e) {// grrr. fileItem.write throws this
+        } catch (FileUploadException e) {
```

Comment:
They probably shouldn't have been caught in this catch-all net. As this is just a Stapler call I wouldn't worry too much about it.